src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,342 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
                         : F.Block(F.Return(invocation));
         }
     }
+
+    internal class SynthesizedAsIntValueMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsIntValueMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi, DiagnosticBag diagnostics, CSharpSyntaxNode node)
+            : base(module.SourceModule, pi,
+                   module.Compilation.GetSpecialType(SpecialType.System_Boolean),
+                   PrivateImplementationDetails.AsIntValueName)
+        {
+            this._module = module;
+            var compilation = _module.Compilation;
+            this.SetParameters(ImmutableArray.Create<ParameterSymbol>(
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Object), 0, RefKind.None, "o"),
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Int32), 1, RefKind.Out, "value")
+                ));
+            Binder.ReportUseSiteDiagnostics(this.ReturnType, diagnostics, node);
+            Binder.ReportUseSiteDiagnostics(this.Parameters[0].Type, diagnostics, node);
+            Binder.ReportUseSiteDiagnostics(this.Parameters[1].Type, diagnostics, node);
+        }
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            ///// <summary>
+            ///// If o is of an integral numeric type and contains a value that is in the range of an int,
+            ///// return true and set the value parameter to that value. Otherwise it leaves an arbitrary
+            ///// value in value and returns false.
+            ///// </summary>
+            //internal static bool AsIntValue(object o, out int value)
+            //{
+            //    value = 0;
+            //    if (o == null)
+            //    {
+            //        return false;
+            //    }
+            //    Type t = o.GetType();
+            //    // CONSIDER: Can we use a switch on t.GetTypeCode() if that method exists?
+            //    if (t == typeof(byte)) { value = (byte)o; return true; }
```

Comment:
I also wonder why we are comparing runtime type objects. It seems it would be cheaper to do
```C#
            if (o is int) { value = (int)o;  return true; }
            . . . 
```